src/main/java/net/education/kyivstar/threads/CustomerRunnable.java

```diff
@@ -0,0 +1,18 @@
+package net.education.kyivstar.threads;
+
+import java.util.concurrent.TimeUnit;
+
+public class CustomerRunnable implements Runnable {
+
+    @Override
+    public void run() {
+        for (int i = 0; i < 10; i++) {
+            System.out.println(Thread.currentThread().getName()+">>>>> " + i);
+            try {
+                TimeUnit.SECONDS.sleep(1);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

Comment:
Your current handling of simply printing the stack trace is generally not advisable, as it doesn't respect the interruption policy and can lead to issues such as threads not terminating when they should.
Change this approach everywhere.

Reply:
Will this approach respect interruption policy 
            try {
                TimeUnit.SECONDS.sleep(1);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
                logger.info("Error "+e);
            }

Reply:
Yes, the approach you suggested will respect the interruption policy. By re-interrupting the current thread using Thread.currentThread().interrupt(), you ensure that the thread's interrupted status is set again, which can be checked by the thread or other parts of the program that rely on it.
`Thread.currentThread().interrupt();`
This line sets the thread's interrupt status again, which was cleared when InterruptedException was caught. This ensures that any subsequent checks of Thread.interrupted() or methods that throw InterruptedException will behave correctly.